packages/forms/frontend/sirius-components-forms/src/propertysections/ButtonPropertySection.tsx

```diff
@@ -0,0 +1,216 @@
+  GQLUpdateWidgetFocusMutationVariables,
+  GQLUpdateWidgetFocusPayload,
+} from './ButtonPropertySection.types';
+import { getTextDecorationLineValue } from './getTextDecorationLineValue';
+import { PropertySectionLabel } from './PropertySectionLabel';
+
+const useStyle = makeStyles<Theme, ButtonPropertySectionStyleProps>((theme) => ({
+  style: {
+    backgroundColor: ({ backgroundColor }) => (backgroundColor ? backgroundColor : theme.palette.primary.light),
+    color: ({ foregroundColor }) => (foregroundColor ? foregroundColor : 'white'),
+    fontSize: ({ fontSize }) => (fontSize ? fontSize : 'inherit'),
+    fontStyle: ({ italic }) => (italic ? 'italic' : 'inherit'),
+    fontWeight: ({ bold }) => (bold ? 'bold' : 'inherit'),
+    textDecorationLine: ({ underline, strikeThrough }) => getTextDecorationLineValue(underline, strikeThrough),
+    '&:hover': {
+      backgroundColor: ({ backgroundColor }) => (backgroundColor ? backgroundColor : theme.palette.primary.main),
+      color: ({ foregroundColor }) => (foregroundColor ? foregroundColor : 'white'),
+      fontSize: ({ fontSize }) => (fontSize ? fontSize : 'inherit'),
+      fontStyle: ({ italic }) => (italic ? 'italic' : 'inherit'),
+      fontWeight: ({ bold }) => (bold ? 'bold' : 'inherit'),
+      textDecorationLine: ({ underline, strikeThrough }) => getTextDecorationLineValue(underline, strikeThrough),
+    },
+  },
+  icon: {
+    marginRight: ({ iconOnly }) => (iconOnly ? theme.spacing(0) : theme.spacing(2)),
+  },
+}));
+
+export const pushButtonMutation = gql`
+  mutation pushButton($input: PushButtonInput!) {
+    pushButton(input: $input) {
+      __typename
+      ... on ErrorPayload {
+        message
+      }
+    }
+  }
+`;
+
+export const updateWidgetFocusMutation = gql`
+  mutation updateWidgetFocus($input: UpdateWidgetFocusInput!) {
+    updateWidgetFocus(input: $input) {
+      __typename
+      ... on ErrorPayload {
+        message
+      }
+    }
+  }
+`;
+
+const isErrorPayload = (payload: GQLPushButtonPayload | GQLUpdateWidgetFocusPayload): payload is GQLErrorPayload =>
+  payload.__typename === 'ErrorPayload';
+
+/**
+ * Defines the content of a Button property section.
+ * The content is submitted when pushing the button.
+ */
+export const ButtonPropertySection = ({
+  editingContextId,
+  formId,
+  widget,
+  subscribers,
+  readOnly,
+}: ButtonPropertySectionProps) => {
+  const props: ButtonPropertySectionStyleProps = {
+    backgroundColor: widget.style?.backgroundColor ?? null,
+    foregroundColor: widget.style?.foregroundColor ?? null,
+    fontSize: widget.style?.fontSize ?? null,
+    italic: widget.style?.italic ?? null,
+    bold: widget.style?.bold ?? null,
+    underline: widget.style?.underline ?? null,
+    strikeThrough: widget.style?.strikeThrough ?? null,
+    iconOnly: widget.buttonLabel ? false : true,
+  };
+  const classes = useStyle(props);
+
+  const httpOrigin: ServerContextValue = useContext(ServerContext);
```

Comment:
`const { httpOrigin } = useContext(ServerContext);`

I'm pretty sure that this would not work